### PR TITLE
rework render and diff logic

### PR DESCRIPTION
**Pull request by dysfunc**

### CODE REVIEW

File: src/diff/diffChildrenByKey.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        parentNode.replaceChild(createElement(newChildren[i]), parentNode.children[i]);
+      }
+    } else if (oldKeys.length > newKeys.length) {
+      for (let i = 0; i < size; i++) {
+        const key = diffedKeys[i];
+
+        for (let a = 0, b = parentNode.children.length; a < b; a++) {
+          const node = parentNode.children[a];
+
+          if (node && node.attributes.key.value === key) {
```
Comment (janmarkuslanger) [inline src/diff/diffChildrenByKey.js]: Could we add a comment here `node.attributes.key.value`  or assign it to a shorter var?

**Comment by janmarkuslanger** [comment]
What is missing here until we can merge this PR

**Comment by dysfunc** [comment]
There's an issue with removal of non-keyed children. I'm not convinced this is a sound change yet - I wanted another pair of eyes to look at these changes in detail

**Comment by dysfunc** [comment]
you're alive! 🎉 and sure 😆

**Comment by janmarkuslanger** [comment]
@dysfunc Sorry i needed to break!

**Pull request closed without merging**